R/benchmark-results.R

```diff
@@ -13,60 +51,52 @@ benchmark_results <- function(
     run_id = NULL,
     batch_id = NULL,
     run_reason = NULL,
-    limit = NULL,
-    days = NULL,
+    name = NULL,
+    earliest_timestamp = NULL,
+    latest_timestamp = NULL,
     simplifyVector = TRUE,
     flatten = TRUE,
     ...) {
-  ## Assert that run_id is a vector of length 5 or less
-  if (length(unique(run_id)) > 5) {
-    stop("Too many run_ids, please limit to 5 or less or consider making separate requests.", call. = FALSE)
-  }
-
-  ## Assert that only one of run_id, batch_id or run_reason is non-NULL
-  non_null_count <- Filter(Negate(is.null), list(run_id, batch_id, run_reason))
-
-  if (length(non_null_count) >= 2) {
-    stop("Only one of run_id, batch_id or run_reason can be non-NULL", call. = FALSE)
-  }
-
-  req <- req_url_path_append(conbench_request(), "benchmark-results")
-
-  if (!is.null(run_reason)) {
-    req <- req_url_query(req, run_reason = run_reason)
-  }
-
-  if (!is.null(days)) {
-    days <- integer_messager(days, label = "days")
-    req <- req_url_query(req, days = days)
-  }
-
-  if (!is.null(limit) && !is.null(run_reason)) {
-    limit <- integer_messager(limit, label = "limit")
-    req <- req_url_query(req, limit = limit)
-  }
-  
-  if (!is.null(limit) && is.null(run_reason)) {
-    stop("you are setting a limit without setting run_reason. please set run_reason and try again", call. = TRUE)
-  }
-
-  if (!is.null(batch_id)) {
-    req <- req_url_query(req, batch_id = paste0(batch_id, collapse = ","))
-  }
-
-  if (!is.null(run_id)) {
-    req <- req_url_query(req, run_id = paste0(run_id, collapse = ","))
+  ## Assert that run_id is a string
+  if (length(run_id) > 1) {
+    stop("Too many run_ids, please limit to 1.", call. = FALSE)
   }
 
+  req <- .construct_request(
+    run_id = run_id,
+    batch_id = batch_id,
+    run_reason = run_reason,
+    name = name,
+    earliest_timestamp = earliest_timestamp,
+    latest_timestamp = latest_timestamp,
+    cursor = NULL
+  )
   resp <- conbench_perform(req)
+  json <- resp_body_json(resp, simplifyVector = simplifyVector, flatten = flatten, ...)
```

Comment:
Oh, I assumed it still worked because this test is still passing:
https://github.com/conbench/conbenchcoms/blob/d8f6aa16c58acd27b3eb04c53908cd627f00e9cc/tests/testthat/test-benchmark-results.R#L39-L46

Oh right - that's not the pagination run ID. I didn't test that. Haha.

But I agree that `simplifyVector` and `flatten` don't make sense in this interface. I could get rid of them.

Reply:
> i'd be interested if @alistaire47 sees any downside to just always returning a tibble (as has been brought up here before: #4)

Vs a vanilla data.frame, no. Since it's stricter about certain things you need to make sure you're not using any of the antipatterns data.frame allows, but I'm pretty sure we're not. And in some cases there's a dep question, but that's moot here.

Having a way to get the raw responses can be handy sometimes, but I'd wait till you need it to refactor.

Reply:
Great. Ok @austin3dickey I am happy to remove `simplifyVector` and `flatten` in a separate PR if you like. But whatever happens I think we have consensus here 🥳